src/azure-cli/azure/cli/command_modules/rdbms/tests/latest/test_rdbms_flexible_commands.py

```diff
@@ -467,3 +468,448 @@ def _test_flexible_server_replica_mgmt(self, database_engine, resource_group):
+                         JMESPathCheck('subnetId',
+                                       '/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/virtualNetworks/{}/subnets/{}'.format(
+                                           self.get_subscription_id(), resource_group, vnet_name_2, 'Subnet' + servers[1][6:])),
+                         JMESPathCheck('host', '{}.{}.database.azure.com'.format(servers[1], database_engine))])
+
+        # flexible-server show to validate delegation is added to both the created server
+        show_result_1 = self.cmd('{} flexible-server show -g {} -n {}'
+                                 .format(database_engine, resource_group, servers[0])).get_output_in_json()
+        show_result_2 = self.cmd('{} flexible-server show -g {} -n {}'
+                                 .format(database_engine, resource_group, servers[1])).get_output_in_json()
+        self.assertEqual(show_result_1['delegatedSubnetArguments']['subnetArmResourceId'],
+                         '/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/virtualNetworks/{}/subnets/{}'.format(
+                             self.get_subscription_id(), resource_group, vnet_name, subnet_name_1))
+        self.assertEqual(show_result_2['delegatedSubnetArguments']['subnetArmResourceId'],
+                         '/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Network/virtualNetworks/{}/subnets/{}'.format(
+                             self.get_subscription_id(), resource_group, vnet_name_2, 'Subnet' + servers[1][6:]))
+
+        # delete all servers
+        self.cmd('{} flexible-server delete -g {} -n {} --yes'.format(database_engine, resource_group, servers[0]),
+                 checks=NoneCheck())
+        self.cmd('{} flexible-server delete -g {} -n {} --yes'.format(database_engine, resource_group, servers[1]),
+                 checks=NoneCheck())
+
+        time.sleep(15 * 60)
+
+        # remove delegations from all vnets
+        self.cmd('network vnet subnet update -g {} --name {} --vnet-name {} --remove delegations'.format(resource_group,
+                                                                                                         subnet_name_1,
+                                                                                                         vnet_name))
+        self.cmd('network vnet subnet update -g {} --name {} --vnet-name {} --remove delegations'.format(resource_group,
+                                                                                                         'Subnet' + servers[1][6:],
+                                                                                                         vnet_name_2))
+
+        # remove all vnets
+        self.cmd('network vnet delete -g {} -n {}'.format(resource_group, vnet_name))
+        self.cmd('network vnet delete -g {} -n {}'.format(resource_group, vnet_name_2))
```

Comment:
With VirtualNetworkPreparer, this part is not needed.

Reply:
In our case - there is an additional step before the vnets can be deleted which is to remove the delegations( Line 667 above). Will try and see if that is taken care off if we use the VirtualNetworkPreparer()

Reply:
> With VirtualNetworkPreparer, this part is not needed.

In our case - there is an additional step before the vnets can be deleted which is to remove the delegations( Line 667 above). Will try and see if that is taken care off if we use the VirtualNetworkPreparer()